frontend/src/main/scala/bloop/bsp/ProjectUris.scala

```diff
@@ -0,0 +1,25 @@
+package bloop.bsp
+
+import java.net.URI
+
+import bloop.Project
+import bloop.engine.{Dag, State}
+import bloop.io.AbsolutePath
+
+object ProjectUris {
+  def getProjectDagFromUri(projectUri: String, state: State): Option[Project] = {
+    if (projectUri.isEmpty) None
+    else {
+      val uri = new URI(projectUri)
+      val query = uri.getQuery().split("&").map(_.split("="))
+      query.headOption.flatMap {
+        case Array("id", projectName) => state.build.getProjectFor(projectName)
+        case _ => None
+      }
+    }
+  }
+
+  def toUri(projectBaseDir: AbsolutePath, id: String): URI = {
+    new URI(s"file:///${projectBaseDir.syntax}?id=${id}")
+  }
```

Comment:
Is this Windows friendly? Do you know the rules to make something Windows-friendly?

Reply:
I removed one `/` and now it is windows friendly. However, this should not be important since this URI is only used to uniquely identify every workspace, the user is not supposed to parse it.